internal/cmd/testenv/main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/kr/pretty"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/internal/testsetup"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/aws"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/azure"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/gcp"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql/core"
+	polaris_log "github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/log"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	checkOnly := func() bool {
+		cleanup := flag.Bool("cleanup", false, "Perform cleanup tasks post CI")
+		precheck := flag.Bool("precheck", false, "Check pre-requirements for running CI tests")
+		flag.Parse()
+		if (!*cleanup && !*precheck) ||
+			(*cleanup && *precheck) {
+			flag.Usage()
+			os.Exit(1)
+		}
+		return *precheck
+	}()
+
+	ctx := context.Background()
+
+	// Load configuration and create client
+	polAccount, err := polaris.DefaultServiceAccount(true)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err := polaris.NewClient(ctx, polAccount, &polaris_log.StandardLogger{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var g errgroup.Group
+
+	// AWS
+	g.Go(func() error {
+		awsAccount, err := client.AWS().Account(ctx, aws.ID(aws.Default()), core.FeatureAll)
+		if checkOnly {
+			switch {
+			case err == nil:
+				return fmt.Errorf("found pre-existing AWS account: %v", pretty.Sprint(awsAccount))
+			case !errors.Is(err, graphql.ErrNotFound):
+				return fmt.Errorf("failed to check AWS account: %v", err)
+			}
+			return nil
+		}
+
+		// cleanup phase
+		switch {
+		case errors.Is(err, graphql.ErrNotFound):
+			return nil
+		case err != nil:
+			return fmt.Errorf("failed to check AWS account: %v", err)
+		}
+		testAcc, err := testsetup.AWSAccount()
+		if err != nil {
+			log.Fatal(err)
+		}
+		if awsAccount.NativeID != testAcc.AccountID {
+			return fmt.Errorf("existing AWS account %q isn't expected test account %q, won't remove",
+				awsAccount.NativeID, testAcc.AccountID)
+		}
+		// TODO: we might need to iterate over awsAccount.Features to remove
+		// all of them in the future
+		return client.AWS().RemoveAccount(ctx, aws.Default(), core.FeatureCloudNativeProtection, false)
+	})
+
+	// Azure
+	g.Go(func() error {
+		azureAccs, err := client.Azure().Subscriptions(ctx, core.FeatureAll, "")
```

Comment:
Is there a reason to not ask for the specific subscription? Something like this:
```
testSub, err := testsetup.AzureSubscription()

...

azureAcc, err := client.Subscription(ctx azure.SubscriptionID(testSub.SubscriptionID), feature core.Feature)
```

Reply:
Good point! Updated